pkg/cmd/cli/cmd/edit.go

```diff
@@ -211,7 +211,11 @@ func RunEdit(fullName string, f *clientcmd.Factory, out io.Writer, cmd *cobra.Co
 			if err != nil {
 				return err
 			}
-			updated, err := resource.NewHelper(info.Client, info.Mapping).Update(info.Namespace, info.Name, false, data)
+			namespace, err := info.ResourceMapping().Namespace(obj)
```

Comment:
Isn't `info.Namespace` the *new* one, in case the user edited the namespace? That's the root of this issue.